zebra-network/src/protocol/external/message.rs

```diff
@@ -352,7 +352,11 @@ where
             // use specific varieties of `RejectReason`.
             ccode: RejectReason::Other,
 
-            reason: e.source().unwrap().to_string(),
+            reason: if let Some(reason) = e.source() {
+                reason.to_string()
+            } else {
+                String::from("No reason detail")
```

Comment:
Since this is a bitcoin network message, we might want to just send an empty string here.

Can you check what `zcashd` does, or what the bitcoin reference says?

Reply:
The bitcoin reference does not seems to say anything about what to do when there is no reason text: https://en.bitcoin.it/wiki/Protocol_documentation#reject

In zcash i am not sure of what they are doing either: https://github.com/zcash/zcash/blob/1b5ab4a06e52c23aca48afb1ebb72a25ef5dd3ca/src/main.cpp#L7334-L7338

I think an empty string can be a bit better than what we have now however i don't have any proof for one side or the other.

Reply:
Changed to empty string in https://github.com/ZcashFoundation/zebra/pull/4609/commits/8a571e5791e77710c067dff8acc3275c05f11fe0

Reply:
Thanks for checking this, I think it's fine for now.